assets/js/data/payment-methods/thunks.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * Internal dependencies
+ */
+import { useEmitResponse } from '../../base/context/hooks/use-emit-response';
+import { emitEventWithAbort } from '../../base/context/event-emit';
+import { EMIT_TYPES } from '../../base/context/providers/cart-checkout/payment-methods/event-emit';
+import type { emitProcessingEventType } from './types';
+import { CART_STORE_KEY } from '../cart';
+
+const { isErrorResponse, isFailResponse, isSuccessResponse, noticeContexts } =
+	useEmitResponse(); // eslint-disable-line react-hooks/rules-of-hooks
+
+// TODO: `useEmitResponse` is not a react hook, it just exposes some functions as
+// properties of an object. Refactor this to not be a hook, we could simply import
+// those functions where needed
+
+/**
+ * Emit the payment_processing event
+ */
+export const emitProcessingEvent: emitProcessingEventType = (
+	currentObserver,
+	setValidationErrors
+) => {
+	// TODO: Fix this type after we move to validation store
+	return ( { dispatch, registry } ) => {
+		const { createErrorNotice, removeNotice } =
+			registry.dispatch( 'core/notices' );
+		removeNotice( 'wc-payment-error', noticeContexts.PAYMENTS );
+		emitEventWithAbort(
+			// TODO Move all this block into a function in thunks & Create buttons to trigger this event (see Alex's PR)
```

Comment:
Will [my testing notes](https://gist.github.com/opr/72f3513feb9e39d95c3db57fada31122) cover this?